homeassistant/components/refoss/sensor.py

```diff
@@ -0,0 +1,194 @@
+        ),
+        RefossSensorEntityDescription(
+            key="current",
+            name="Current",
+            device_class=SensorDeviceClass.CURRENT,
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+            subkey="current",
+        ),
+        RefossSensorEntityDescription(
+            key="factor",
+            name="Power Factor",
+            device_class=SensorDeviceClass.POWER_FACTOR,
+            state_class=SensorStateClass.MEASUREMENT,
+            subkey="factor",
+        ),
+        RefossSensorEntityDescription(
+            key="energy",
+            name="This Month Energy",
+            device_class=SensorDeviceClass.ENERGY,
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+            subkey="mConsume",
+        ),
+        RefossSensorEntityDescription(
+            key="energy_returned",
+            name="This Month Energy Returned",
+            device_class=SensorDeviceClass.ENERGY,
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+            subkey="mConsume",
+        ),
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Refoss device from a config entry."""
+
+    @callback
+    def init_device(coordinator):
+        """Register the device."""
+        device = coordinator.device
+
+        if not isinstance(device, ElectricityXMix):
+            return
+        descriptions = SENSORS.get(device.device_type)
+        new_entities = []
+        for channel in device.channels:
+            for description in descriptions:
+                entity = RefossSensor(
+                    coordinator=coordinator,
+                    channel=channel,
+                    description=description,
+                )
+                new_entities.append(entity)
+
+        async_add_entities(new_entities)
+
+    for coordinator in hass.data[DOMAIN][COORDINATORS]:
+        init_device(coordinator)
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(hass, DISPATCH_DEVICE_DISCOVERED, init_device)
+    )
+
+
+class RefossSensor(RefossEntity, SensorEntity):
+    """Refoss Sensor Device."""
+
+    entity_description: RefossSensorEntityDescription
+    _uom: UnitOfMeasurement | None = None
+
+    def __init__(
+        self,
+        coordinator: RefossDataUpdateCoordinator,
+        channel: int,
+        description: RefossSensorEntityDescription,
+    ) -> None:
+        """Init Refoss sensor."""
+        super().__init__(coordinator, channel)
+        self.entity_description = description
+        self._attr_unique_id = f"{super().unique_id}{description.key}"
+        device_type = coordinator.device.device_type
+        self._attr_name = (
```

Comment:
Please adopt your code to use translatable names. More info can be found in the developers docs

Reply:
https://developers.home-assistant.io/blog/2023/03/27/entity_name_translations/

Reply:
Here, I referred to Shelly's.

Reply:
Adding translations for new entities is required. With translation placeholders (https://developers.home-assistant.io/blog/2024/01/19/entity-translations-placeholders/), it should be easy to add support for them.

> Here, I referred to Shelly's.

The shelly integration where added before we had translation support therefore they have more time to migrate to translations

Reply:
thanks